### PR TITLE
update auto-random.md

### LINKED ISSUE #17826: auto_random(5,54) 超出分配范围
File: [/release-6.5/auto-random.md](https://docs.pingcap.com/zh/tidb/v6.5/auto-random)
image
如官网所述，AUTO_RANDOM(5, 54) 代替 AUTO_RANDOM(5)，使得 TiDB 不会分配出大于 9007199254740991 (2^53-1) 的整数。

版本：5.7.25-TiDB-v6.5.0
我们的表结构:
image
CREATE TABLE `xxxxxxx_record` (
  `id` bigint(20) unsigned NOT NULL /*T![auto_rand] AUTO_RANDOM(5, 54) */ COMMENT '主键id',
) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_bin /*T![auto_rand_base] AUTO_RANDOM_BASE=59350326110818 */ COMMENT='测试表'

测试插入后：
image
9007199254741021
9007199254741024
9007199254741033
9007199254741162
9007199254741166
9007199254741243
9007199254741282
结果：生成数据很多都是大于9007199254740991 (2^53-1) 数据

期待回复感谢

### PR DESCRIPTION
**Pull request by tangenta**
### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### What is changed, added or deleted? (Required)

Refine the description for reserved bits. Close #17826.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v8.2 (TiDB 8.2 versions)
- [x] v8.1 (TiDB 8.1 versions)
- [x] v8.0 (TiDB 8.0 versions)
- [x] v7.6 (TiDB 7.6 versions)
- [x] v7.5 (TiDB 7.5 versions)
- [x] v7.1 (TiDB 7.1 versions)
- [x] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Review by lcwangchao — APPROVED** [review]

File: auto-random.md [hunk 1]
```diff
@@ -79,16 +79,23 @@ TiDB 自动分配的 `AUTO_RANDOM(S, R)` 列值共有 64 位：
...
+
+| 保留位      | 分片位 | 自增位     |
+|-------------|--------|------------|
+| (64-R) bits | S bits | (R-S) bits |
+
+- 是否有符号位取决于该列是否存在 `UNSIGNED` 属性。
 - 保留位的长度为 `64-R`，保留位的内容始终为 `0`。
 - 分片位的内容通过计算当前事务的开始时间的哈希值而得。要使用不同的分片位数量（例如 10），可以在建表时指定 `AUTO_RANDOM(10)`。
 - 自增位的值保存在存储引擎中，按顺序分配，每次分配完值后会自增 1。自增位保证了 `AUTO_RANDOM` 列值全局唯一。当自增位耗尽后，再次自动分配时会报 `Failed to read auto-increment value from storage engine` 的错误。
+- 关于取值范围：最终生成值包含的最大位数 = 分片位 + 自增位。有符号位的列的取值范围是 `[-(2^(R-1))+1, (2^(R-1))-1]`。有符号位的列的取值范围是 `[-(2^R)+1, (2^R)-1]`。
```
Comment (tangenta) [inline auto-random.md:98]:
```suggestion
- 关于取值范围：最终生成值包含的最大位数 = 分片位 + 自增位。有符号位的列的取值范围是 `[-(2^(R-1))+1, (2^(R-1))-1]`。无符号位的列的取值范围是 `[-(2^R)+1, (2^R)-1]`。
```

File: auto-random.md [hunk 2]
```diff
@@ -79,16 +79,23 @@ TiDB 自动分配的 `AUTO_RANDOM(S, R)` 列值共有 64 位：
...
+
+| 保留位      | 分片位 | 自增位     |
+|-------------|--------|------------|
+| (64-R) bits | S bits | (R-S) bits |
+
+- 是否有符号位取决于该列是否存在 `UNSIGNED` 属性。
 - 保留位的长度为 `64-R`，保留位的内容始终为 `0`。
 - 分片位的内容通过计算当前事务的开始时间的哈希值而得。要使用不同的分片位数量（例如 10），可以在建表时指定 `AUTO_RANDOM(10)`。
 - 自增位的值保存在存储引擎中，按顺序分配，每次分配完值后会自增 1。自增位保证了 `AUTO_RANDOM` 列值全局唯一。当自增位耗尽后，再次自动分配时会报 `Failed to read auto-increment value from storage engine` 的错误。
+- 关于取值范围：最终生成值包含的最大位数 = 分片位 + 自增位。有符号位的列的取值范围是 `[-(2^(R-1))+1, (2^(R-1))-1]`。无符号位的列的取值范围是 `[-(2^R)+1, (2^R)-1]`。
```
Comment (tangenta) [inline auto-random.md:98]:
```suggestion
- 关于取值范围：最终生成值包含的最大位数 = 分片位 + 自增位。有符号位的列的取值范围是 `[-(2^(R-1))+1, (2^(R-1))-1]`。无符号位的列的取值范围是 `[0, (2^R)-1]`。
```

**Review by qiancai — APPROVED** [review]
Rest LGTM

File: auto-random.md [hunk 3]
```diff
@@ -79,16 +79,23 @@ TiDB 自动分配的 `AUTO_RANDOM(S, R)` 列值共有 64 位：
...
+
+| 保留位      | 分片位 | 自增位     |
+|-------------|--------|------------|
+| (64-R) bits | S bits | (R-S) bits |
+
+- 是否有符号位取决于该列是否存在 `UNSIGNED` 属性。
 - 保留位的长度为 `64-R`，保留位的内容始终为 `0`。
 - 分片位的内容通过计算当前事务的开始时间的哈希值而得。要使用不同的分片位数量（例如 10），可以在建表时指定 `AUTO_RANDOM(10)`。
 - 自增位的值保存在存储引擎中，按顺序分配，每次分配完值后会自增 1。自增位保证了 `AUTO_RANDOM` 列值全局唯一。当自增位耗尽后，再次自动分配时会报 `Failed to read auto-increment value from storage engine` 的错误。
+- 关于取值范围：最终生成值包含的最大位数 = 分片位 + 自增位。有符号位的列的取值范围是 `[-(2^(R-1))+1, (2^(R-1))-1]`。无符号位的列的取值范围是 `[0, (2^R)-1]`。
```
[the remark this answers is not in the document]
Comment (tangenta) [reply auto-random.md:98]: 应该是 -127 到 -1

**Review by hfxsd — APPROVED** [review]

**Pull request merged**